baseplate/integration/pyramid.py

```diff
@@ -86,6 +86,45 @@ class ServerSpanInitialized(BaseplateEvent):
     pass
 
 
+class BaseplateHeaderTrustHandler(object):
+
+    """Class used by BaseplateConfigurator to validate trace and edge context headers. Specifically
+    implements hmac validation for the trace headers.
+
+    param bool trust_trace_headers: Whether to trust trace headers by default
+    param bool trust_edge_context_header: Whether to trust the edge context header by default
+    param baseplate.secrets.SecretsStore secret_store: secret store which should contain a
+        versioned secret at the path trace/hmac if doing hmac validation of the trace headers.
+    """
+    def __init__(
+        self,
+        trust_trace_headers=False,
+        trust_edge_context_header=False,
+        secret_store=None
+    ):
+        self.trust_trace_headers = trust_trace_headers
+        self.trust_edge_context_header = trust_edge_context_header
+        self.secret_store = secret_store
+
+    def should_trust_trace_headers(self, request):
+        if self.secret_store:
```

Comment:
Related to the other comments, I don't think we should have branching logic based on the existence of the secret store. Classes that need the secret store should concretely depend on them, catastrophically failing if it doesn't exist.